Build error with Python 3.7

I get the following error with Python 3.7.0 on Arch Linux:

```
make -j8
Scanning dependencies of target libuavcan_dsdlc
[  1%] Running dsdl compiler
Compiler failure
Traceback (most recent call last):
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 300, in enum_last_value
    last = next(it)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 107, in run_generator
    text = generate_one_type(template_expander, t)
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 239, in generate_one_type
    text = template_expander(t=t)  # t for Type
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 307, in expand
    return template(**args)
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1030, in __call__
    u = "".join(self._render(self.parsetree, self.current_data))
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1130, in render
    output.extend(self.render(elem[2], data))
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1104, in render
    output.append(unicode(_eval(elem[1], data)))
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1079, in _eval
    return self.evalfunc(expr, data)
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 884, in eval
    x = eval(self.compile(expr), {"__builtins__":self.eval_allowed_globals}, locals)
  File "", line 1, in <module>
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1030, in __call__
    u = "".join(self._render(self.parsetree, self.current_data))
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1104, in render
    output.append(unicode(_eval(elem[1], data)))
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1079, in _eval
    return self.evalfunc(expr, data)
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 884, in eval
    x = eval(self.compile(expr), {"__builtins__":self.eval_allowed_globals}, locals)
  File "", line 1, in <module>
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1030, in __call__
    u = "".join(self._render(self.parsetree, self.current_data))
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1138, in render
    output.extend(self.render(elem[1], data))
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/pyratemp.py", line 1119, in render
    for i in loop_iter:
RuntimeError: generator raised StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdlc", line 61, in <module>
    dsdlc_run(args.source_dir, args.incdir, args.outdir)
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 62, in run
    run_generator(types, output_dir)
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 111, in run_generator
    die(ex)
  File "/home/julianoes/src/Firmware/src/modules/uavcan/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 89, in die
    raise DsdlCompilerException(str(text))
libuavcan_dsdl_compiler.DsdlCompilerException: generator raised StopIteration
generator raised StopIteration
make[2]: *** [libuavcan/CMakeFiles/libuavcan_dsdlc.dir/build.make:160: libuavcan_dsdlc_run.stamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:129: libuavcan/CMakeFiles/libuavcan_dsdlc.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

Same as https://github.com/PX4/Firmware/issues/10644

--- a/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py
+++ b/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
 #
 # UAVCAN DSDL compiler for libuavcan
 #

--- a/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py
+++ b/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py
@@ -40,11 +40,11 @@ def run(source_dirs, include_dirs, output_dir):
     possibly empty list of search directories (containing DSDL definition files that can be referenced from the types
     that are going to be parsed), and the output directory path (possibly nonexistent) where the generated C++
     header files will be stored.
-    
+
     Note that this module features lazy write, i.e. if an output file does already exist and its content is not going
     to change, it will not be overwritten. This feature allows to avoid unnecessary recompilation of dependent object
     files.
-    
+
     Args:
         source_dirs    List of root namespace directories to parse.
         include_dirs   List of root namespace directories with referenced types (possibly empty). This list is

--- a/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py
+++ b/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py
@@ -298,7 +298,10 @@ def make_template_expander(filename):
         def enum_last_value(iterable, start=0):
             it = iter(iterable)
             count = start
-            last = next(it)
+            try:
+                last = next(it)
+            except StopIteration:
+                return
             for val in it:
                 yield count, False, last
                 last = val

